File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> positives;
        vector<int> negatives;

        for(int i = 0 ; i < nums.size();i++){
            if(nums[i]>=0)positives.push_back(nums[i]);
            else negatives.push_back(nums[i]);
        }
        int i = 0,j = 0,k=0;
        while(k<nums.size()){
            nums[k++] = positives[i++];
            nums[k++] = negatives[j++];
        }

        return nums;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        ans = [0]*len(nums)

        i = 0
        j = 1

        for k in range(len(nums)):
            if nums[k] > 0:
                ans[i]= nums[k]
                i+=2
            else:
                ans[j] = nums[k]
                j+=2

        return ans
